x-pack/plugins/observability_solution/observability_ai_assistant_app/server/rule_connector/index.ts

```diff
@@ -189,6 +195,24 @@ If available, include the link of the conversation at the end of your answer.`
     ),
   };
 
+  const hasSlackConnector = !!connectorsList.filter(
+    (connector) => connector.actionTypeId === '.slack'
+  ).length;
+
+  if (hasSlackConnector) {
+    const slackConnectorInstruction: AdHocInstruction = {
```

Comment:
can you also check for the presence of the `execute_connector` function? In some cases it's not available (eg when the LLM has exceeded the # of allowed function calls).

Reply:
Thanks for the review @dgieselaar 
I've updated the code.